transform/gc.go

```diff
@@ -8,7 +8,7 @@ import (
 
 // MakeGCStackSlots converts all calls to runtime.trackPointer to explicit
 // stores to stack slots that are scannable by the GC.
-func MakeGCStackSlots(mod llvm.Module) bool {
+func MakeGCStackSlots(mod llvm.Module, shouldHiveGlobals bool) bool {
```

Comment:
There is a typo here.

```suggestion
func MakeGCStackSlots(mod llvm.Module, shouldHideGlobals bool) bool {
```

However, would it work to always disable this global? If it works for WASI, it will likely work for JS too. The global was never meant to be exposed. Making it universally hidden or internal avoids a lot of complexity and simplifies this PR.
```